src/Compilers/Core/Portable/CommandLine/CommandLineArguments.cs

```diff
@@ -510,16 +510,22 @@ public IEnumerable<AnalyzerReference> ResolveAnalyzerReferences(IAnalyzerAssembl
                 }
             };
 
-            var resolvedReferences = ArrayBuilder<AnalyzerFileReference>.GetInstance();
+            var resolvedReferences = PooledHashSet<AnalyzerFileReference>.GetInstance();
```

Comment:
Can't use a hash set here because it introduces non-determinism into the compilation. Think we need two collections here: the resolved list and the hash set.